designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -122,14 +185,161 @@ class UsageStatisticsReportsSettingsDeterminer(
     }
   }
 
-  private[statistics] def determineQueryParams(): Future[Map[String, String]] =
-    for {
+  private[statistics] def determineQueryParams(): Future[Map[String, String]] = {
+    val generalStats = for {
       scenariosInputData <- fetchNonArchivedScenariosInputData()
       scenariosStatistics = scenariosInputData.map(determineStatisticsForScenario).combineAll.mapValuesNow(_.toString)
       fingerprint <- fingerprintSupplier(config, nuFingerprintFileName)
       basicStatistics = determineBasicStatistics(fingerprint, config)
     } yield basicStatistics ++ scenariosStatistics
 
+    val queryParamsFromScenarios = fetchNonArchivedScenariosInputData()
+      .map {
+        case List() => Map.empty[String, String]
+        case scenariosInputData =>
+          val scenarios = scenariosInputData.count(!_.isFragment)
+
+          //        Nodes stats
+          val nodes        = scenariosInputData.map(_.nodes).sorted
+          val nodesMedian  = nodes.get(nodes.length / 2).getOrElse(0)
+          val nodesAverage = nodes.sum / scenariosInputData.length
+
+          //        Category stats
+          val categories = scenariosInputData.map(_.processCategory).toSet.size
+
+          //        Version stats
+          val versions        = scenariosInputData.map(_.processVersion.value).sorted
+          val versionsMedian  = versions.get(versions.length / 2).getOrElse(-1)
+          val versionsAverage = versions.sum / scenariosInputData.length
+
+          //        Author stats
+          val authors = scenariosInputData.map(_.createdBy).toSet.size
+
+          //        Fragment stats
+          val fragments        = scenariosInputData.map(_.fragmentsUsed).sorted
+          val fragmentsMedian  = fragments.get(fragments.length / 2).getOrElse(0)
+          val fragmentsAverage = fragments.sum / scenarios
+
+          Map(
+            "m_n" -> nodesMedian,
+            "v_n" -> nodesAverage,
+            "c"   -> categories,
+            "v_m" -> versionsMedian,
+            "v_v" -> versionsAverage,
+            "a_n" -> authors,
+            "m_f" -> fragmentsMedian,
+            "v_f" -> fragmentsAverage,
+          ).mapValuesNow(_.toString)
+      }
+    val activityQueryParams = fetchActivity()
+      .map {
+        case List()           => Map.empty[String, String]
+        case listOfActivities =>
+          //        Attachment stats
+          val attachments = listOfActivities
+            .map { activity =>
+              activity.attachments
+            }
+            .map(_.length)
+            .sorted
+          val attachmentMedian  = attachments.get(attachments.length / 2).getOrElse(0)
```

Comment:
I'm wondering if we needs median for the attachments or whether they exist. Maybe we could leave only average and instead of exists, we could calculate total sum?